src/Plutarch/Test/QuickCheck/Instances.hs

```diff
@@ -0,0 +1,974 @@
+
+-- | @since 2.0.0
+instance PCoArbitrary PPositive where
+    pcoarbitrary = pcoarbitrary . liftTestable pto
+
+-- | @since 2.0.0
+instance PArbitrary PRational where
+    parbitrary = do
+        (TestableTerm x) <- parbitrary
+        (TestableTerm y) <- parbitrary
+        return $ TestableTerm $ pcon $ PRational x y
+
+    pshrink (TestableTerm x) =
+        [ TestableTerm $ pcon $ PRational a b
+        | (TestableTerm a) <- shrink (TestableTerm $ pnumerator # x)
+        , (TestableTerm b) <- shrink (TestableTerm $ pdenominator # x)
+        ]
+
+instance PCoArbitrary PRational where
+    pcoarbitrary (TestableTerm x) = pcoarbitrary n . pcoarbitrary d
+      where
+        n = TestableTerm $ pnumerator # x
+        d = TestableTerm $ pdenominator # x
+
+-- | @since 2.0.0
+instance PArbitrary PString where
+    parbitrary = pconstantT . T.pack <$> arbitrary
+    pshrink = fmap (pconstantT . T.pack) . shrink . T.unpack . pliftT
+
+instance PCoArbitrary PString where
+    pcoarbitrary = coarbitrary . T.unpack . pliftT
+
+-- | @since 2.0.0
+instance PArbitrary a => PArbitrary (PMaybe a) where
+    parbitrary = do
+        (TestableTerm x) <- parbitrary
+        frequency [(3, return $ TestableTerm $ pcon $ PJust x), (1, return $ pconT PNothing)]
+    pshrink (TestableTerm x)
+        | plift $ pisJust # x =
+            TestableTerm (pcon PNothing) :
+                [ TestableTerm $ pcon $ PJust a
+                | (TestableTerm a) <- shrink (TestableTerm $ pfromJust # x)
+                ]
+        | otherwise = []
+
+instance PCoArbitrary a => PCoArbitrary (PMaybe a) where
+    pcoarbitrary (TestableTerm x)
+        | plift $ pisJust # x = variant 1 . pcoarbitrary (TestableTerm $ pfromJust # x)
+        | otherwise = variant 0
+
+-- | @since 2.0.0
+instance (PIsData a, PArbitrary a) => PArbitrary (PMaybeData a) where
+    parbitrary = do
+        (TestableTerm x) <- parbitrary
+        elements
+            [ TestableTerm $ pcon $ PDJust $ pdcons @"_0" # pdata x # pdnil
+            , TestableTerm $ pcon $ PDNothing pdnil
+            ]
+    pshrink (TestableTerm x)
+        | plift $ pisDJust # x =
+            pconT (PDNothing pdnil) :
+                [ TestableTerm $ pcon $ PDJust $ pdcons @"_0" # pdata a # pdnil
+                | (TestableTerm a) <- shrink (TestableTerm $ pfromDJust # x)
+                ]
+        | otherwise = []
+
+instance (PIsData a, PCoArbitrary a) => PCoArbitrary (PMaybeData a) where
+    pcoarbitrary (TestableTerm x)
+        | plift $ pisDJust # x = variant 1 . pcoarbitrary (TestableTerm $ pfromDJust # x)
+        | otherwise = variant 0
+
+isRight = flip pmatchT $ \case
+    PRight _ -> pconstant True
+    _ -> pconstant False
+pright = flip pmatchT $ \case
+    PRight a -> a
+    _ -> ptraceError "asked for PRight when it is PLeft"
+pleft = flip pmatchT $ \case
+    PLeft a -> a
+    _ -> ptraceError "asked for PLeft when it is PRight"
```

Comment:
Are these meant to have no signatures? Did they get detached from somewhere else?